apps/asap-server/src/migrations/1637748102-move-type-proposal-to-grant-research.ts

```diff
@@ -0,0 +1,67 @@
+/* istanbul ignore file */
+import { RestResearchOutput, Results, Squidex } from '@asap-hub/squidex';
+import { ResearchOutputSubtype } from '@asap-hub/model';
+
+import { Migration } from '../handlers/webhooks/webhook-run-migrations';
+
+interface Fetcher {
+  (pointer: number): Promise<Results<RestResearchOutput>>;
+}
+
+interface ItemProcessor {
+  (researchOutput: RestResearchOutput): unknown;
+}
+
+const fetchSquidex =
+  (
+    squidexClient: Squidex<RestResearchOutput>,
+    type: 'Proposal' | 'Grant Document',
+  ) =>
+  (pointer: number) =>
+    squidexClient.fetch({
+      $top: 10,
+      $skip: pointer,
+      $orderby: 'created asc',
+      $filter: `data/type/iv eq '${type}'`,
+    });
+
+const iterate = async (fetcher: Fetcher, process: ItemProcessor) => {
+  let pointer = 0;
+  let result: Results<RestResearchOutput>;
+
+  do {
+    result = await fetcher(pointer);
+
+    for (const researchOutput of result.items) {
+      await process(researchOutput);
+    }
+
+    pointer += 10;
+  } while (pointer < result.total);
+};
+
+export default class MoveResearchOutputTextToDescription extends Migration {
+  up = async (): Promise<void> => {
+    const squidexClient = new Squidex<RestResearchOutput>('research-outputs');
+
+    await iterate(fetchSquidex(squidexClient, 'Proposal'), (researchOutput) =>
```

Comment:
you are building something that has already been built:  https://github.com/yldio/asap-hub/blob/709b1e48126d9665e9970c68f45d1407c2b551e5/apps/asap-server/src/migrations/1622036244-add-default-onboarded-field-to-users.ts#L8

and im pretty sure that it doesn't iterate through all the records because many have been left out.

Although your way does not fetch all the records, just the proposals. Maybe expand the `applyToAllItemsInCollection` to accept a filter? Or iterate over all of them but skip the ones that aren't proposal programatically?